core/Plugin/Controller.php

```diff
@@ -269,27 +269,36 @@ protected function renderView(ViewInterface $view)
      *                           '@$pluginName/myTemplate.twig'. Alternatively you can include the full path:
      *                           '@anyOtherFolder/otherTemplate'. The trailing '.twig' is not needed.
      * @param array $variables   For instance array('myViewVar' => 'myValue'). In template you can use {{ myViewVar }}
+     * @param string|null $viewType 'basic' or 'admin'. If null, determined based on the controller instance type.
      * @return string
      * @since 2.5.0
      * @api
      */
-    protected function renderTemplate($template, array $variables = array())
+    protected function renderTemplate($template, array $variables = array(), $viewType = null)
     {
         if (false === strpos($template, '@') || false === strpos($template, '/')) {
             $template = '@' . $this->pluginName . '/' . $template;
         }
 
         $view = new View($template);
 
+        $this->checkViewType($viewType);
+
+        if (empty($viewType)) {
+            $viewType = $this instanceof ControllerAdmin ? 'admin' : 'basic';
+        }
+
         // alternatively we could check whether the templates extends either admin.twig or dashboard.twig and based on
         // that call the correct method. This will be needed once we unify Controller and ControllerAdmin see
         // https://github.com/piwik/piwik/issues/6151
-        if ($this instanceof ControllerAdmin) {
-            $this->setBasicVariablesView($view);
-        } elseif (empty($this->site) || empty($this->idSite)) {
-            $this->setBasicVariablesView($view);
+        if ($this instanceof ControllerAdmin && $viewType == 'admin') {
+            $this->setBasicVariablesView($view, $viewType);
+        } elseif ((empty($this->site) || empty($this->idSite))
+            && $viewType == 'admin'
+        ) {
+            $this->setBasicVariablesView($view, $viewType);
```

Comment:
@diosmosis wouldn't we also call this when using `viewType=view` in some cases? basic variables does different things than general variables?

Reply:
Yes seems so, I'm not sure why I made this change... I'll have to test again